api/api.py

```diff
@@ -7,7 +7,60 @@ class Blob:
 
    To wrap a string as a blob, call ``blob(my_str)``"""
 
-def docker_build(ref: str, context: str, build_args: Dict[str, str] = {}, dockerfile: str = "Dockerfile", dockerfile_contents: Union[str, Blob] = "") -> None:
+class LiveUpdateStep:
+  """A step in the process of performing a LiveUpdate on an image's container.
+
+  See the LiveUpdate documentation, coming soon!
+  """
+  pass
+
+def fall_back_on(files: Union[str, List[str]]) -> LiveUpdateStep:
+  """Specify that any changes to the given files will cause Tilt to *fall back* to a
+  full image build (rather than performing a live update).
+
+  ``fall_back_on`` step(s) may only go at the beginning of your list of steps.
+
+  (Files must be a subset of the files that we're already watching for this image;
+  that is, if any files fall outside of DockerBuild.context or CustomBuild.deps,
+  an error will be raised.)
+
+  Args:
+      files: a string or list of strings of files. If relative, will be evaluated relative to the Tiltfile. Tilt compares these to the local paths of edited files when determining whether to fall back to a full image build.
+  """
+  pass
+
+def sync(local_path: str, remote_path: str) -> LiveUpdateStep:
+  """Specify that any changes to `localPath` should be synced to `remotePath`
+
+  May not follow any `run` steps in a `live_update`.
+
+  Args:
+      localPath: A path relative to the Tiltfile's directory. Changes to files matching this path will be synced to `remotePath`.
+          Can be a file (in which case just that file will be synced) or directory (in which case any files recursively under that directory will be synced).
+      remotePath: container path to which changes will be synced. Must be absolute.
+  """
+  pass
+
+def run(cmd: str, trigger: Union[List[str], str] = []) -> LiveUpdateStep:
+  """Specify that the given `cmd` should be executed when updating an image's container
+
+  May not precede any `sync` steps in a `live_update`.
+
+  Args:
+    cmd: A shell command.
+    trigger: If the ``trigger`` argument is specified, the build step is only run on changes to the given file(s). Paths relative to Tiltfile.
```

Comment:
```suggestion
    trigger: If the ``trigger`` argument is specified, the build step is only run when there are changes to the given file(s). Paths relative to Tiltfile.
```

(my first read of the old wording was like "oh it'll somehow intelligently only run go build on the files that changed - wait how can it do that?")